src/libraries/System.Private.CoreLib/src/System/Math.cs

```diff
@@ -133,6 +133,30 @@ public static decimal Abs(decimal value)
             return decimal.Abs(value);
         }
 
+        [Intrinsic]
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static double Abs(double value)
+        {
+            // Implementation based on https://git.musl-libc.org/cgit/musl/tree/src/math/fabs.c
```

Comment:
This comment is a bit "misleading", IMO. But, I also don't know whether or not its actually required.

This `Abs` implementation isn't really unique/special. It's the implementation the JIT is already using, as well as MSVC, GCC, Clang, Musl, etc; because on x86/x64 its the "natural" way to write this. `Abs` for a one's complement value involves toggling the sign-bit and for raw instructions; that's effectively `x & ~(-0.0)` which then gets optimized down to `x & 0x7FFFFFFF` so that `andps reg, [addr]` can be emitted on x86/x64.

Reply:
+1 for removing that comment, The algorithm to do ABS is pretty simple and obvious to require any reference

Reply:
This comment is not unique in this (Math.cs) file though :wink:

Reply:
Definitely, but many of the other implementations weren't pre-existing or almost globally implemented by other compilers.